src/charm.py

```diff
@@ -481,134 +43,61 @@ def __init__(self, *args) -> None:  # type: ignore[no-untyped-def]
             args: Variable list of positional arguments passed to the parent constructor.
         """
         super().__init__(*args)
-        self.framework.observe(self.on.config_changed, self._on_config_changed_with_warning)
+        if not self.unit.is_leader():
+            self.unit.status = WaitingStatus(
+                "follower unit is idling, "
+                "remove follower units using `juju scale-application {self.app.name} 1`"
```

Comment:
Perhaps change this to something along "this charm only supports a single unit, please remove the additional units"

Reply:
Updated, thanks!